source/kmip.rst

```diff
@@ -0,0 +1,75 @@
+.. _kmip:
+
+Using the Key Management Interoperability Protocol (KMIP) 
+============================================================
+
+This feature is **technical preview** quality.
+
+|PSMDB| adds support for secure transfer of keys using the `OASIS Key Management Interoperability Protocol (KMIP) <https://docs.oasis-open.org/kmip/kmip-spec/v2.0/os/kmip-spec-v2.0-os.html>`__. The KMIP implementation was tested with the `PyKMIP server <https://pykmip.readthedocs.io/en/latest/server.html>`__ and the `HashiCorp Vault Enterprise KMIP Secrets Engine <https://www.vaultproject.io/docs/secrets/kmip>`__.
+
+KMIP enables the communication between key management systems and the database server. KMIP can perform the following operations:
+
+* Streamline encryption key management
+* Eliminate redundant key management processes
+
+.. admonition:: KMIP parameters
+
+   .. list-table::
+      :widths: 25 25 15 35
+      :header-rows: 1
+   
+      * - Option
+        - Type
+        - Description
+      * - --kmipServerName
+        - string
+        - The hostname or IP address of the KMIP server.
+      * - --kmipPort
+        - number
+        - The port used to communicate with the KMIP server. 
+      * - --kmipServerCAFile
+        - string
+        - The path to the TLS certificate file. CA file is used to validate secure client connection to the KMIP server.
+      * - --kmipClientCertificateFile
+        - string
+        - The path to the client certificate file. The client certificate file is used by MongoDB to authenticate to the KMIP server. 
+      * - --kmipClientKeyFile
+        - string
+        - The path to the KMIP CA key.
+      * - --kmipKeyIdentifier
+        - string
+        - The unique KMIP identifier for an existing key within the KMIP server. MongoDB uses the key associated with the identifier as the system key. You can only use this parameter when enabling encryption for the **mongod** instance for the first time. If the ``--kmipKeyIdentifier`` parameter is not specified, MongoDB will request KMIP server to create a new key to use as the system key. If the KMIP server cannot find a key with the specified identifier or the data is already encrypted with a key, MongoDB throws an error.
```

Comment:
PSMDB works differently.

* The parameter is required, users have to specify a key name
* if it doesn't exist, the server will create a key on the KMIP server with the specified identifier